metadata-ingestion/src/datahub/ingestion/source/sql_queries.py

```diff
@@ -156,68 +167,90 @@ def get_report(self) -> SqlQueriesSourceReport:
+        self,
+    ) -> Iterable[Union[MetadataWorkUnit, MetadataChangeProposalWrapper]]:
         logger.info(f"Parsing queries from {os.path.basename(self.config.query_file)}")
+
+        with self.report.new_stage("Collecting queries from file"):
+            queries = list(self._parse_query_file())
+            logger.info(f"Collected {len(queries)} queries for processing")
+
+        with self.report.new_stage("Processing queries through SQL parsing aggregator"):
+            for query_entry in queries:
+                self._add_query_to_aggregator(query_entry)
+
+        with self.report.new_stage("Generating metadata work units"):
+            logger.info("Generating workunits from SQL parsing aggregator")
+            yield from self.aggregator.gen_metadata()
+
+    def _parse_query_file(self) -> Iterable["QueryEntry"]:
+        """Parse the query file and yield QueryEntry objects."""
         with open(self.config.query_file) as f:
             for line in f:
                 try:
                     query_dict = json.loads(line, strict=False)
                     entry = QueryEntry.create(query_dict, config=self.config)
-                    yield from self._process_query(entry)
+                    self.report.num_queries_parsed += 1
+                    if self.report.num_queries_parsed % 1000 == 0:
+                        logger.info(f"Parsed {self.report.num_queries_parsed} queries")
+                    yield entry
                 except Exception as e:
-                    logger.warning("Error processing query", exc_info=True)
-                    self.report.report_warning("process-query", str(e))
+                    self.report.num_queries_parse_failures += 1
+                    self.report.warning(
+                        title="Error processing query",
+                        message="Query skipped due to parsing error",
+                        context=line.strip(),
+                        exc=e,
+                    )
 
-        logger.info("Generating workunits")
-        yield from self.builder.gen_workunits()
+    def _add_query_to_aggregator(self, query_entry: "QueryEntry") -> None:
+        """Add a query to the SQL parsing aggregator."""
+        try:
+            observed_query = ObservedQuery(
+                query=query_entry.query,
+                timestamp=query_entry.timestamp,
+                user=query_entry.user,
+                session_id=query_entry.session_id,
+                default_db=self.config.default_db,
+                default_schema=self.config.default_schema,
+            )
 
-    def _process_query(self, entry: "QueryEntry") -> Iterable[MetadataWorkUnit]:
-        self.report.num_queries_parsed += 1
-        if self.report.num_queries_parsed % 1000 == 0:
-            logger.info(f"Parsed {self.report.num_queries_parsed} queries")
+            # If we have explicit lineage, use it directly
+            if query_entry.upstream_tables or query_entry.downstream_tables:
```

Comment:
the observed_query object should be constructed inside the else clause

Reply:
nice catch! addressed in 9be3598b141bb76481db07dd440d64c1caeb940f